broker-core/src/main/java/io/zeebe/broker/job/JobState.java

```diff
@@ -277,8 +281,11 @@ static State forValue(byte value) {
     }
   }
 
+  /** <b>Note:</b> calling this method will reset the payload of the job record */
```

Comment:
I would just rename the method to `resetPayloadAndUpdateJob` or something instead of putting all these warnings, imo, instead of all these comment warnings